code/frameworks/react-vite/src/preset.ts

```diff
@@ -51,5 +51,5 @@ export const viteFinal: NonNullable<StorybookConfig['viteFinal']> = async (confi
     );
   }
 
-  return config;
+  return { ...config, plugins };
```

Comment:
before, this preset would discard any config set by previous `viteFinal` presets. We never noticed it because this would always run first in Storybook. But in the Vitest plugin it appears that `main.ts` runs first, so we need to handle it properly here.